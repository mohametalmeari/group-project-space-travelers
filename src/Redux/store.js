import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './features/missions/missionSlice';

const store = configureStore({
  reducer: {
    mission: missionReducer,
  },
});

export default store;
