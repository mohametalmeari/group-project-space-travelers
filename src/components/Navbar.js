import { NavLink } from 'react-router-dom';
import { Nav, Container } from 'react-bootstrap';
import styles from './Navbar.module.css';

const links = [
  { path: '/', text: 'Rockets' },
  { path: '/missions', text: 'Missions' },
  { path: '/my-profile', text: 'My Profile' },
];

const Navbar = () => (
  <Container>
    <Nav className={styles.navContainer}>
      <h2 className="nav-logo">
        <NavLink className={styles.navlinkContainer} to="/">
          <img
            width="40"
            height="40"
            src="https://img.icons8.com/badges/48/saturn-planet.png"
            alt="saturn-planet"
          />
          Space Travelers Hub
        </NavLink>
      </h2>
      <ul className={`menu ${styles.navLinkRow}`}>
        {links.map((link) => (
          <li className="page-link" key={link.text}>
            <NavLink className="page-link" activeClassName={styles.active} to={link.path}>
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </Nav>
  </Container>
);

export default Navbar;
