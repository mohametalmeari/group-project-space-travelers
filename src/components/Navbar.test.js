import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import Navbar from './Navbar';

describe('Navbar Component', () => {
  test('Test render of Navbar', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
