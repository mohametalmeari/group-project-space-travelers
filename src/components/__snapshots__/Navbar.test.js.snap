// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Test render of Navbar 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="navContainer nav"
    >
      <h2
        class="nav-logo"
      >
        <a
          aria-current="page"
          class="navlinkContainer active"
          href="/"
        >
          <img
            alt="saturn-planet"
            height="40"
            src="https://img.icons8.com/badges/48/saturn-planet.png"
            width="40"
          />
          Space Travelers Hub
        </a>
      </h2>
      <ul
        class="menu"
      >
        <li
          class="page-link"
        >
          <a
            activeclassname="active"
            aria-current="page"
            class="page-link active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li
          class="page-link"
        >
          <a
            activeclassname="active"
            class="page-link"
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li
          class="page-link"
        >
          <a
            activeclassname="active"
            class="page-link"
            href="/my-profile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
